eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -264,26 +262,50 @@ class Setup(datadir: File,
       _ <- if (config.getBoolean("api.enabled")) {
         logger.info(s"json-rpc api enabled on port=${config.getInt("api.port")}")
         implicit val materializer = ActorMaterializer()
-        val api = new Service {
+        val api = if(config.getBoolean("api.use-new-version")){
+          new NewService {
+
+          override val actorSystem = kit.system
 
-          override def scheduler = system.scheduler
+          override val mat = materializer
```

Comment:
why do we need that?

Reply:
The materializer is needed to work with akka's Flow in the websocket handler, the scheduler is used to artificially create a delay when the authentication was rejected.